apps/content/services/admin/asset_recitation_audio_tracks_direct_upload_service.py

```
@@ -109,36 +97,52 @@ def finish_upload(self, key: str, upload_id: str, parts: list[dict[str, Any]]) -
-            logger.warning("Failed to get size for uploaded object %s: %s", r2_key, e)
-            size_bytes = 0
-
-        # Get current track to check if duration_ms was already set
-        track = RecitationSurahTrack.objects.only("duration_ms").get(audio_file=key)
+        # Fall back to head_object if size_bytes not provided or 0
+        if not size_bytes:
+            try:
+                logger.info("backend server are computing size_bytes instead of of parsing it directly from frontend!")
+                head = s3.head_object(Bucket=settings.CLOUDFLARE_R2_BUCKET, Key=r2_key)
+                size_bytes = int(head.get("ContentLength", 0))
+            except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)